### PR TITLE
[Doc] Repin samples off the deleted ray-ml 2.46.0.0e19ea tags

### LINKED ISSUE #5123: [Bug] ray-job.batch-inference.yaml pins a rayproject/ray-ml tag that returns 404
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

Others

### What happened + What you expected to happen

`ray-operator/config/samples/ray-job.batch-inference.yaml` pins a container image that doesn't exist, so the sample can't run as written. Line 16:

```yaml
image: rayproject/ray-ml:2.46.0.0e19ea-py39-gpu
```

That tag returns 404 from Docker Hub:

```console
$ curl -s -o /dev/null -w "%{http_code}\n" \
    https://hub.docker.com/v2/repositories/rayproject/ray-ml/tags/2.46.0.0e19ea-py39-gpu
404
```

Applying the sample leaves the head pod in `ImagePullBackOff`, so the RayJob never starts.

Present on both `master` and `v1.6.0`.

The underlying cause is that `rayproject/ray-ml` is no longer published for public consumption. Versioned tags stop at `2.30.0` — `2.31.0` through `2.56.1` all return 404 — and Ray pushed a family of 16 `2.46.0.deprecated*` tags on 2025-05-07. So this isn't a one-off typo in a build hash; there is no working `ray-ml` tag at 2.46.0 to correct it to.

Picking a replacement needs a decision rather than a find-and-replace, because the sample's workload genuinely needs ML libraries that the base `rayproject/ray` images don't ship. The mounted `sample_code.py` imports `transformers` and `PIL` and runs a Hugging Face `image-classification` pipeline on GPU. Two options I can see:

1. Pin `rayproject/ray:<current>-gpu` and add a `pip install transformers pillow torch` step, either in the RayJob's `runtimeEnvYAML` or as a `RUN` layer in a small custom image.
2. Build and publish a KubeRay-side sample image that carries the dependencies, and pin that.

Happy to open a PR for whichever direction maintainers prefer.

### Downstream: the Ray docs quote this file, and quote it inconsistently

`ray-project/ray`'s [`rayjob-batch-inference-example.md`](https://github.com/ray-project/ray/blob/master/doc/source/cluster/kubernetes/examples/rayjob-batch-inference-example.md) tells the reader to `curl` this sample from `v1.6.0` and then quotes the head-node spec back to them — but shows `rayproject/ray-ml:2.6.3-gpu`, not the tag actually in the file. So a reader sees one image in the tutorial prose and pulls another. (`ray-ml:2.6.3-gpu` does still resolve, which makes the mismatch easy to miss.)

Both halves need fixing, but this one is the upstream half: correcting the Ray doc's quote first would just make it accurately quote a broken tag. Once this sample lands on a working image I'll sync the Ray-side snippet to match.

### Reproduction script

```bash
kubectl apply -f https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-job.batch-inference.yaml
kubectl get pods -w   # head pod enters ImagePullBackOff
```

Tag existence alone:

```bash
curl -s -o /dev/null -w "%{http_code}\n" \
  https://hub.docker.com/v2/repositories/rayproject/ray-ml/tags/2.46.0.0e19ea-py39-gpu
```

### Anything else

Found while auditing container image pins across the Ray documentation. AI assistance (Claude Code) was used to inventory the pins and verify each tag against Docker Hub; I reviewed and confirmed the findings above.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jang-hs**
## Why are these changes needed?

The `rayproject/ray-ml:2.46.0.0e19ea-*` tag family was deleted from Docker Hub, so five samples land in `ImagePullBackOff` as written:

```console
$ curl -s -o /dev/null -w "%{http_code}\n" \
    https://hub.docker.com/v2/repositories/rayproject/ray-ml/tags/2.46.0.0e19ea-py39-gpu
404
```

Affected files (every remaining `2.46.0.0e19ea` pin in the repo):

- `ray-operator/config/samples/ray-job.batch-inference.yaml`
- `ray-operator/config/samples/ray-service.mobilenet.yaml`
- `ray-operator/config/samples/ray-service.stable-diffusion.yaml`
- `ray-operator/config/samples/pytorch-resnet-image-classifier/ray-job.pytorch-image-classifier.yaml`
- `ray-operator/config/samples/pytorch-text-classifier/ray-job.pytorch-distributed-training.yaml`

`ray-ml` itself is still published, just under `.deprecated`-suffixed tags, with `2.49.2.deprecated-*` as the newest release family. So this is a repin, not a rebuild, and it is the same fix #4811 already applied to `ray-service.text-summarizer.yaml`: bump image and `rayVersion` to 2.49.2, keeping each sample's existing Python and CPU/GPU variant.

Note for #5123: the issue reasoned that no working `ray-ml` tag existed, so a replacement image had to be built or the ML dependencies moved into `runtimeEnvYAML`. The `.deprecated` tag family makes that unnecessary — the samples' dependencies still come from the image (see below).

## Related issue number

Closes #5123

## Labels

- [ ] If this PR has user-facing changes that require documentation updates at release time, I have added the `doc-updates-required` label.
- [ ] If this PR contains breaking changes, I have added the `breaking-change` label.

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
  - [ ] Unit tests
  - [x] Manual tests

### Manual test instructions

**1. Every image tag referenced by the five changed files resolves on Docker Hub.**

```console
$ for f in ray-operator/config/samples/ray-job.batch-inference.yaml \
          ray-operator/config/samples/ray-service.mobilenet.yaml \
          ray-operator/config/samples/ray-service.stable-diffusion.yaml \
          ray-operator/config/samples/pytorch-resnet-image-classifier/ray-job.pytorch-image-classifier.yaml \
          ray-operator/config/samples/pytorch-text-classifier/ray-job.pytorch-distributed-training.yaml; do
    grep -oE 'rayproject/[a-z-]+:[^ ]+' "$f" | while read -r img; do
      printf "%s %s\n" \
        "$(curl -s -o /dev/null -w '%{http_code}' "https://hub.docker.com/v2/repositories/${img%%:*}/tags/${img##*:}")" "$img"
    done
  done
200 rayproject/ray-ml:2.49.2.deprecated-py39-gpu
200 rayproject/ray-ml:2.49.2.deprecated-py39-cpu
200 rayproject/ray-ml:2.49.2.deprecated-py39-cpu
200 rayproject/ray-ml:2.49.2.deprecated-py39-gpu
200 rayproject/ray-ml:2.49.2.deprecated-py39-gpu
200 rayproject/ray:2.49.2
200 rayproject/ray-ml:2.49.2.deprecated-py39
200 rayproject/ray-ml:2.49.2.deprecated-py39
200 rayproject/ray-ml:2.49.2.deprecated-py311-gpu
```

The same loop on `master` returns 404 for all seven `ray-ml` lines.

**2. The images really are Ray 2.49.2 on the expected Python**, so the new `rayVersion` is accurate. From the image config of `ray-ml:2.49.2.deprecated-py39-gpu`:

```
COPY .whl/ray-2.49.2-cp39-cp39-manylinux2014_x86_64.whl .
CUDA_VERSION=12.1.1
```

**3. The suffix-less tag is the GPU image**, so `pytorch-image-classifier`'s GPU worker keeps a CUDA image despite `2.49.2.deprecated-py39` carrying no `-gpu`:

```
2.49.2.deprecated-py39      sha256:21b0b684...  CUDA 12.1.1
2.49.2.deprecated-py39-gpu  sha256:21b0b684...  (same digest)
2.49.2.deprecated-py39-cpu  sha256:5e6be059...  (no CUDA)
```

Same aliasing holds on the older still-live family (`2.30.0-py39` == `2.30.0-py39-gpu`), so this is the tag layout's convention rather than a one-off.

**4. The samples' image-supplied dependencies survive the bump.** `requirements_compiled.txt` at `ray-2.49.2` still pins `tensorflow==2.15.1` (py<3.12, needed by `mobilenet`), `torch==2.3.0`, `transformers==4.36.2`, `pillow==10.3.0` (needed by `batch-inference`). Keeping each sample on its current Python variant rather than moving to py311 is deliberate: the py39 images are the ones that carry TensorFlow under that marker.

**5. `batch-inference`'s mounted `sample_code.py` needs no change.** It passes `compute=ActorPoolStrategy(...)` to `map_batches`; in Ray 2.49.2 `get_compute_strategy` takes the legacy branch and only logs a deprecation warning ([util.py#L605](https://github.com/ray-project/ray/blob/ray-2.49.2/python/ray/data/_internal/util.py#L605)).

**Not covered:** full end-to-end runs. `batch-inference` and `stable-diffusion` need 4x T4 GPUs and `pytorch-image-classifier` needs a GKE GCSFuse-backed service account, none of which I have access to. None of these five samples are in the `test/sampleyaml` matrix, so CI does not exercise them either.

### CODE REVIEW

**Comment by jang-hs** [comment]
The one red check, `test-history-server-e2e-nightly-operator`, is unrelated to this PR. This PR changes five sample YAMLs under `ray-operator/config/samples/`; that job runs `historyserver/test/e2e`, which does not read any of them.

From the job's `e2e-log.tar` artifact, only `TestCollector` failed, and it failed waiting on MinIO rather than on anything under test:

```
--- RUN   TestCollector
    s3.go:31: [12:53:30Z] Successfully applied ../../config/minio.yaml to namespace minio-dev
    s3.go:62:
        Timed out after 300.262s.
        MinIO API endpoint should be ready
        ...
        caused by: Get "http://localhost:9000/": dial tcp 127.0.0.1:9000: connect: connection refused
### FAIL: TestCollector (303.58s)
```

`TestHistoryServer` then applied the same `minio.yaml` eight minutes later in the same run and connected immediately, and passed:

```
    s3.go:31: [13:01:34Z] Successfully applied ../../config/minio.yaml to namespace minio-dev
    s3.go:63: [13:01:34Z] Port-forwarded MinIO API port to localhost:9000 successfully
### PASS: TestHistoryServer (1217.65s)
```

Same commit, same cluster, same manifest — so the MinIO setup itself is fine and this is a flake. The job passed on the last 14 master commits.

One note in case it is useful beyond this PR: `connection refused` (not a timeout) for the full 5 minutes suggests the `kubectl port-forward` from `EnsureS3Client` never bound or died, and the surrounding `Eventually` only retries `ListBuckets` — it never re-establishes the forward, so once it is gone the wait is guaranteed to burn the whole `TestTimeoutMedium`. Retrying the port-forward inside that loop would turn this class of failure into a short retry instead of a 5-minute hard fail. Happy to open a separate issue for it rather than mixing it into this PR.

I have rebased onto master to re-trigger CI.

**Comment by jang-hs** [comment]
@win5923 @kevin85421 ping, green since Aug 11.

@win5923, you +1'd option 1 on #5123 the day after I opened this, so to be clear this PR does neither option. ray-ml is still published under .deprecated tags, newest is 2.49.2.deprecated-*, so the five samples are just an image repin - same fix #4811 already used for ray-service.text-summarizer.yaml.

The catch is those tags aren't maintained and Ray stops publishing at 2.50.x, so it fixes the samples now but isn't permanent. Happy to redo it as option 1 if you prefer that.
